src/SecretManager/Microsoft.DncEng.SecretManager.ScenarioTests/WrappedTokenProvider.cs

```
@@ -7,9 +10,37 @@ public sealed class WrappedTokenProvider : ITokenCredentialProvider
     {
         private readonly TokenCredential _tokenCredential;
 
+        /// <inheritdoc/>
+        public string ApplicationId { get; internal set; }
+        /// <inheritdoc/>
+        public string TenantId { get; internal set; }
+
         public WrappedTokenProvider(TokenCredential tokenCredential)
         {
             _tokenCredential = tokenCredential;
+            SetCredentialIdentityValues();
+        }
+
+        /// <inheritdoc/>
+        internal void SetCredentialIdentityValues()
+        {
+            try
+            {
+                // Get a token from the credential provider
+                var tokenRequestContext = new TokenRequestContext(new[] { "https://management.azure.com/.default" });
+                var token = _tokenCredential.GetToken(tokenRequestContext, CancellationToken.None);
+
+                // Decode the JWT to get user identity information
+                var handler = new JwtSecurityTokenHandler();
+                var jsonToken = handler.ReadToken(token.Token) as JwtSecurityToken;
+                ApplicationId = jsonToken?.Claims?.FirstOrDefault(claim => claim.Type == "oid")?.Value ?? "Claim Oid Not Found";
+                TenantId = jsonToken?.Claims?.FirstOrDefault(claim => claim.Type == "tenant_id")?.Value ?? "Claim tenant_id Not Found";
+            }
+            catch
```

Catch-everything seems heavy-handed. Specifically, this will hide a failure to actually get a token (e.g. permissions problems) and it seems that should be bubbled up to the rest of the app.

---

Then the process is no longer 'non-volatile' it is now a requirement that valid credentials can be extracted using this code. 

That is not desirable here. We don't want this logging operation or any of the required data gathering efforts to cause the actual process to fail.

---

I remove the guidance message from the ID values on a failure and replaced that with comments explaining why we want this process to be non-volatile.

---

Based on our offline conversation, I will remove the notes in the ID values when claims are not found and then resolve this item.